### DISCUSSION TITLE
Absence of shipments data in getShopReceipt

### DISCUSSION THREAD

**Discussion by teraxis** [etsy/open-api#1061] [Ideas]
I'm making a request to get the recipe data using the getShopReceipt method. According to the documentation I should receive such array:
`"shipments": [
{
"receipt_shipping_id": 1,
"shipment_notification_timestamp": 946684800,
"carrier_name": "string",
"tracking_code": "string"
}
],`

In Etsy seller cabinet this data is exist, but in getShopReceipt I getting only:

` {"receipt_id":1316170236, ... "shipments":[{}], ... }`

or

` {"receipt_id":1316170236, ... "shipments":[{},{}], ... }`

I get same empty data for all reciepts.

**Comment by Gareth064** [comment]
I'm nearly certain that "Shipments" was part of the old v2 API.

In V3, you find shipping related data against the transaction attached to the Receipt.

Receipt > transactions[] > transaction > shipping method (other stuff)

**Reply by teraxis** [reply]
[shipping_profile_id] => 52342955378
[min_processing_days] => 5
[max_processing_days] => 10
[**shipping_method] =>** 
[shipping_upgrade] => 
[expected_ship_date] => 1528385924
[buyer_coupon] => 0
[shop_coupon] => 0

And there is no tracking ID

**Reply by Gareth064** [reply]
Has the order been shipped?

**Reply by teraxis** [reply]
![AnyDesk_9byid6qZ7L](https://github.com/etsy/open-api/assets/38819145/62d4fc27-808b-4fc6-83d5-001bc5a86f8a)

**Reply by Gareth064** [reply]
Interesting. Let me see what data I get back from a shipped order. I will post back shortly.

**Reply by Gareth064** [reply]
Ok, you were right the first time. V3 Does have "Shipments" array on the Receipt object

![image](https://github.com/etsy/open-api/assets/22307223/a4b35338-5f62-4860-bdd6-8342f4b85b94)

And for me, I have this populated for an order which was shipped

![image](https://github.com/etsy/open-api/assets/22307223/e21d4a2e-aba1-40ac-adec-0a6d76b9df19)

From GUI

![image](https://github.com/etsy/open-api/assets/22307223/226e94a6-4ba7-4bc8-a1c7-59f55a4b0760)

**Reply by teraxis** [reply]
Thank you! I found the problem. I use the rhysnhall/etsy-php-sdk library in etsy-php-sdk/src/Resource.php a function that converts the result:
  public function toArray() {
    $array = [];
    $properties = get_object_vars($this)['_properties'];
    foreach($properties as $property => $value) {
      if(is_object($value)) {
        if($value instanceof Resource) {
          $array[$property] = $value->toArray();
        }
        else {
          $array[$property] = get_object_vars($value);
        }
      }
      else {
        $array[$property] = $value;
      }
    }
    return $array;
  }
  
  but it returns a corrupted array

**Reply by teraxis** [reply]
stdClass Object is protected

    [shipments] => Array
        (
            [0] => Etsy\Resources\Shipment Object
                (
                    [_associations:protected] => Array
                        (
                        )

                    [_rename:protected] => Array
                        (
                        )

                    [_properties:**protected**] => stdClass Object
                        (
                            [receipt_shipping_id] => 1147008188772
                            [shipment_notification_timestamp] => 1685116800
                            [carrier_name] => USPS
                            [tracking_code] => 9205590264287637312135
                        )

                )

        )
